csharp/autobuilder/Semmle.Autobuild.CSharp.Tests/BuildScripts.cs

```diff
@@ -981,8 +981,39 @@ public void TestDotnetVersionWindows()
         {
             actions.RunProcess["cmd.exe /C dotnet --list-sdks"] = 0;
             actions.RunProcessOut["cmd.exe /C dotnet --list-sdks"] = "2.1.3 [C:\\Program Files\\dotnet\\sdks]\n2.1.4 [C:\\Program Files\\dotnet\\sdks]";
-            actions.RunProcess[@"cmd.exe /C powershell -NoProfile -ExecutionPolicy unrestricted -file C:\Project\install-dotnet.ps1 -Version 2.1.3 -InstallDir C:\Project\.dotnet"] = 0;
-            actions.RunProcess[@"cmd.exe /C del C:\Project\install-dotnet.ps1"] = 0;
+            actions.RunProcess[@"cmd.exe /C pwsh -NoProfile -ExecutionPolicy unrestricted -Command ""[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; &([scriptblock]::Create((Invoke-WebRequest -UseBasicParsing 'https://dot.net/v1/dotnet-install.ps1'))) -Version 2.1.3 -InstallDir C:\Project\.dotnet"""] = 0;
```

Comment:
Do I get it correctly that `TestDotnetVersionWindows` and `TestDotnetVersionWindowsNoPwsh` differ only in two lines? If so, I would make it more explicit, and share the common bits.

Reply:
I totally agree that code sharing is good, but for these tests I have a slight preference towards explicitness (and therefor duplication), so that changing otherwise shared bits does not make a test pass which shouldn't have passed. Does that make sense?

Reply:
I'm not sure I understand. Wouldn't the following be equally explicit?
```
public void TestDotnetVersionWindows(Action<TestActions> installDotnet)
{
 ...
 installDotnet(actions);
 ...
}

[Fact]
public void TestDotnetVersionWindowsWithPwsh()
{
  TestDotnetVersionWindows(a =>
  {
    ...
  });
}
```

Reply:
OK, I'll fix :-)